_posts/2018-02-26-grpc-and-deadlines.md

```diff
@@ -78,7 +78,8 @@ if (context->IsCancelled()) {
 
 ```go
 if ctx.Err() == context.Canceled {
```

Comment:
More complete would be:

```go
switch ctx.Err() {
  case context.Canceled:
    return status.New(codes.Canceled, "client cancelled, abandoning.")
  case context.DeadlineExceeded:
    return status.New(codes.DeadlineExceeded, "deadline exceeded")
  case nil:
    break
  default:
    return status.New(codes.Unknown, "unknown context error")
}
```

Hmm, maybe we should add a `status.FromContextError(error) *status.Status` or `status.ConvertContextError(error) error` convenience function that does this, so the user's code is:

```go
if err := ctx.Err(); err != nil {
  return status.FromContextError(error).Err()
}
```

Sadly, until 1.6 support is gone, we'll need 2 conditionally compiled files to do this so we can handle both "context" and "x/net/context" for go1.7 and go1.8.. but it's probably worth it.